Makefile.dune

```diff
@@ -9,6 +9,7 @@
 
 # use DUNEOPT=--display=short for a more verbose build
 # DUNEOPT=--display=short
+export PYTHONIOENCODING=utf-8
```

Comment:
You shouldn't do this because you have no guarantee that Dune will be invoked through this Makefile.

What you want instead is probably this patch:

```diff
diff --git a/doc/dune b/doc/dune
index c82e5a3df4..c5fca82358 100644
--- a/doc/dune
+++ b/doc/dune
@@ -32,7 +32,7 @@
   unreleased.rst
   (env_var SPHINXWARNOPT))
  (action
-  (run env COQLIB=%{project_root} sphinx-build -q %{env:SPHINXWARNOPT=-W} -b html sphinx %{targets})))
+  (run env COQLIB=%{project_root} PYTHONIOENCODING=utf-8 sphinx-build -q %{env:SPHINXWARNOPT=-W} -b html sphinx %{targets})))
 
 (rule
  (targets refman-pdf)
@@ -48,7 +48,7 @@
   (env_var SPHINXWARNOPT))
  (action
   (progn
-   (run env COQLIB=%{project_root} sphinx-build -q %{env:SPHINXWARNOPT=-W} -b latex sphinx %{targets})
+   (run env COQLIB=%{project_root} PYTHONIOENCODING=utf-8 sphinx-build -q %{env:SPHINXWARNOPT=-W} -b latex sphinx %{targets})
    (chdir %{targets} (run make LATEXMKOPTS=-silent)))))
 
 ; Installable directories are not yet fully supported by Dune.  See
```

Reply:
Is there a reasonable way to set the variable for all Dune steps?

Reply:
Updated.

Reply:
> Is there a reasonable way to set the variable for all Dune steps?

Actually, yes. You could do this:

```diff
diff --git a/dune b/dune
index cf7221ce62..732d4b282d 100644
--- a/dune
+++ b/dune
@@ -4,7 +4,8 @@
  (release (flags :standard -rectypes)
           (ocamlopt_flags :standard -O3 -unbox-closures))
  (ireport (flags :standard -rectypes -w -9-27-40+60)
-          (ocamlopt_flags :standard -O3 -unbox-closures -inlining-report)))
+          (ocamlopt_flags :standard -O3 -unbox-closures -inlining-report))
+ (_ (env-vars (PYTHONIOENCODING utf-8))))
 
 ; Information about flags for release mode:
 ;
```

(Corresponding doc: https://dune.readthedocs.io/en/stable/dune-files.html#env)